File: ruby/bob/bob.rb
module Bob
  def self.hey(statement)
    is_blank = blank?(statement)
    is_question = question?(statement)
    is_yell = yell?(statement)

    if is_blank
      "Fine. Be that way!"
    elsif is_question && is_yell
      "Calm down, I know what I'm doing!"
    elsif is_question
      "Sure."
    elsif is_yell
      "Whoa, chill out!"
    else
      "Whatever."
    end
  end

  def self.blank?(text)
    text !~ /[[:graph:]]/
  end

  def self.question?(text)
    text.rstrip.end_with?("?")
  end

  def self.yell?(text)
    text =~ /[[:upper:]]/ && text !~ /[[:lower:]]/
  end
end

File: javascript/bob/bob.js
String.prototype.isBlank = function() {
    return !/\S/.test(this);
}

String.prototype.isQuestion = function() {
    return /\?\s*$/.test(this);
}

String.prototype.isYell = function() {
    return /[A-Z]/.test(this) && !/[a-z]/.test(this);
}

export const hey = (message) => {
    if (message.isBlank())
        return "Fine. Be that way!";
    else if (message.isQuestion() && message.isYell())
        return "Calm down, I know what I'm doing!"
    else if (message.isQuestion())
        return "Sure."
    else if (message.isYell())
        return "Whoa, chill out!"
    else
        return "Whatever."
};
